src/core/server/http/router/response_adapter.ts

```diff
@@ -129,21 +129,21 @@ export class HapiResponseAdapter {
     }
 
     // we use for BWC with Boom payload for error responses - {error: string, message: string, statusCode: string}
-    const error = new Boom('', {
+    const error = new Boom.Boom('', {
       statusCode: kibanaResponse.status,
     });
 
     error.output.payload.message = getErrorMessage(payload);
 
     const attributes = getErrorAttributes(payload);
     if (attributes) {
+      // @ts-expect-error `attributes` does not exist on the `Payload` type, however it does exist in real life (https://github.com/hapijs/boom/issues/277)
       error.output.payload.attributes = attributes;
```

Comment:
After a bit of digging (https://github.com/hapijs/boom/issues/277), it turns out that `payload.attributes` probably isn't used correctly here. In fact I can't really figure out why we set this property as it's not really clear that it's used elsewhere.

@mikecote @restrry @pgayvallet three places in this code-base do we set `error.output.payload.attributes`. Do any of you know why/what it's used for?

Reply:
@watson I think this can be cleaned up now. There's two places I can see it still used ([here](https://github.com/elastic/kibana/blob/700225061a3c736aec36ddc12e9669b213894d3b/src/core/server/saved_objects/import/validate_references.ts#L74) and [here](https://github.com/elastic/kibana/blob/b3d97764a0d942f2561ec33266d3d07f267a57a4/src/core/server/saved_objects/export/get_sorted_objects_for_export.ts#L157)) by the saved objects export API.

From what I recall, this was used to indicate what objects caused the API to return 400 (which may not return 400 errors anymore). I'm not sure if the UI got to leverage this but with the new logic it seems this isn't necessary anymore.

Reply:
Also for insight, the usage goes way back to my PR (https://github.com/elastic/kibana/pull/30326) and more specifically, this commit: https://github.com/elastic/kibana/pull/30326/commits/ec662dbc842f194e8a630697910815da5f5443a1.

Reply:
Great news! Thanks for looking into this 👍 In that case, I'll open a separate PR to get rid of these as it's technically unrelated to this one.

Reply:
After digging into this, it does seem like we still rely on this. So I'll leave it as is